astroid/nodes/node_classes.py

```diff
@@ -3350,20 +3353,21 @@ class TypeAlias(_base_nodes.AssignTypeNode):
     <TypeAlias l.1 at 0x7f23b2e4e198>
     """
 
-    _astroid_fields = ("type_params", "value")
+    _astroid_fields = ("name", "type_params", "value")
+
+    name: AssignName
+    type_params: list[TypeVar | ParamSpec | TypeVarTuple]
+    value: NodeNG
```

Comment:
I'm not totally sure why we're adopting this pattern instead of using the constructor.

Reply:
I don't like it as well, but I do think we should at least try and keep one style so that we can then refactor it to a desired style all at once?

Reply:
No worries. Sometimes it seems like we prefer not to [mass-replace](https://github.com/pylint-dev/astroid/pull/2219#discussion_r1238936145), but not a big deal, let's keep moving. 👍